source/sundials/kinsol.cc

```diff
@@ -258,6 +284,21 @@ namespace SUNDIALS
     status = KINSetMAA(kinsol_mem, data.anderson_subspace_size);
     AssertKINSOL(status);
 
+#  if DEAL_II_SUNDIALS_VERSION_GTE(6, 0, 0)
+    // From the manual: this must be called BEFORE KINInit
+    status = KINSetOrthAA(kinsol_mem, data.anderson_qr_orthogonalization);
```

Comment:
This assumes a particular order of the numbers to translate between the KINSOL numbers and our enum. It would be good if this were documented in the enum declaration.